### PR TITLE
[public images] Minor reorganisation

### PR DESCRIPTION
**Pull request by k-dimple**
1. Remove explanation section since currently it only has 1 topic
 2. Update the generic descriptions to specify Vagrant since currently that is the only image we are talking about
 3. Correct some minor typos
 4. Include pointers to public images and OCI registries in the main public cloud page

### CODE REVIEW

File: public-images/public-images-how-to/index.rst
Comment (mstpn) [inline public-images/public-images-how-to/index.rst:1]: Agree with the flattening here. We can always add headings back when there are more offerings.

File: all-clouds/index.rst [hunk 1]
```diff
@@ -30,10 +30,25 @@ For further details, refer to the cloud-specific documentation:
...
 
-------------------------------------------------------------------------------------------------
+Apart from the specific public clouds, Canonical also produces a variety of Ubuntu images for use in other platforms such as LXD, MAAS and OCI container registries.
+
+
+..  grid:: 1 1 2 2
+   :padding: 0
+
+   .. grid-item-card:: :doc:`Ubuntu on OCI container registries <oci:index>`   
+   .. grid-item-card:: :doc:`Ubuntu public images <public-images:index>`
```
Comment (mstpn) [inline all-clouds/index.rst:46]: Capitalization on Public Images to be in line with our use elsewhere.

File: public-images/public-images-how-to/run-a-vagrant-box.rst [hunk 2]
```diff
@@ -2,7 +2,25 @@
...
+
+.. code::
+
+   vagrant init <BOX>
+
+For upstream, use a call such as:
+
+.. code::
+
+   vagrant init ubuntu/focal64
```
Comment (mstpn) [inline public-images/public-images-how-to/run-a-vagrant-box.rst:15]: As per [John's previous comment](https://github.com/canonical/ubuntu-cloud-docs/pull/214#discussion_r1569235125), lets point to `ubuntu/jammy64`. Something that got missed when this was all one paragraph.

**Review by mstpn — CHANGES_REQUESTED** [review]

File: all-clouds/index.rst (hunk 1 above)
Comment (k-dimple) [reply all-clouds/index.rst:46]: Oh yeah, good catch!

File: public-images/public-images-how-to/run-a-vagrant-box.rst (hunk 2 above)
Comment (k-dimple) [reply public-images/public-images-how-to/run-a-vagrant-box.rst:15]: Done, thanks!

**Review by mstpn — APPROVED** [review]

**Pull request merged**